shared/src/commonMain/kotlin/io.newm.shared/internal/services/CardanoWalletAPI.kt

```diff
@@ -6,24 +6,29 @@ import io.ktor.client.request.get
 import io.ktor.client.request.parameter
 import io.ktor.http.ContentType
 import io.ktor.http.contentType
+import io.ktor.serialization.JsonConvertException
 import io.newm.shared.di.NetworkClientFactory
 import io.newm.shared.public.models.error.KMMException
-import org.koin.core.component.KoinComponent
-import kotlin.coroutines.cancellation.CancellationException
-
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
+import org.koin.core.component.KoinComponent
+import kotlin.coroutines.cancellation.CancellationException
 
 internal class CardanoWalletAPI(networkClient: NetworkClientFactory) : KoinComponent {
 
     private val authClient: HttpClient = networkClient.authHttpClient()
 
     @Throws(KMMException::class, CancellationException::class)
     suspend fun getWalletNFTs(xpub: String): List<List<LedgerAssetMetadata>> =
-        authClient.get("/v1/cardano/nfts") {
-            contentType(ContentType.Application.Json)
-            parameter("xpub", xpub)
-        }.body()
+        try {
+            authClient.get("/v1/cardano/nfts") {
+                contentType(ContentType.Application.Json)
+                parameter("xpub", xpub)
+            }.body()
+        } catch(e: JsonConvertException) {
+            print("Error fetching wallet NFTs: ${e.message}")
+            emptyList()
```

Comment:
I think we might want to revert this change, considering the possibility of errors occurring. In the case of errors, we should display an error state. Conversely, if a wallet genuinely has no NFTs, we should show an empty state.

Reply:
I think this was crashing because of not handling expired refresh tokens...?  But yeah it should be reverted.

Reply:
Should we do a check when they enter their xPub key that it's valid, instead of allowing them to successfully add an invalid key?

Reply:
We can do a check on the scanner, so that we only take the address with the xpub prefix

Reply:
doesn't verify it's an actual wallet though.  we can come back to this.